NFIQ2/NFIQ2Algorithm/src/tool/nfiq2_ui_refresh.cpp

```diff
@@ -768,60 +769,37 @@ NFIQ2UI::parseModel(const NFIQ2UI::Arguments &arguments)
 		modelInfoFilePath = arguments.flags.model;
 	}
 
-	// FIXME: Make below a separate function and make a Model class
-	static const std::string ModelInfoKeyName { "Name" };
-	static const std::string ModelInfoKeyTrainer { "Trainer" };
-	static const std::string ModelInfoKeyDescription { "Description" };
-	static const std::string ModelInfoKeyVersion { "Version" };
-	static const std::string ModelInfoKeyPath { "Path" };
-	static const std::string ModelInfoKeyHash { "Hash" };
-
-	std::unique_ptr<BE::IO::Properties> props;
-	std::string modelFile, hash;
-
-	try {
-		props.reset(new BE::IO::PropertiesFile(
-		    modelInfoFilePath, BE::IO::Mode::ReadOnly));
-	} catch (const BE::Error::Exception &e) {
-		throw NFIQ2UI::PropertyParseError(
-		    "Unable to parse model info file '" + modelInfoFilePath +
-		    "' (" + e.whatString() + ')');
-	}
-
+	std::shared_ptr<NFIQ::ModelInfo> modelInfoObj {};
```

Comment:
This shouldn't be a `shared_ptr`. Making other code changes suggested should make this possible.